baton.lua

```diff
@@ -96,33 +104,55 @@ end
 function Player:_updateControls()
   if not self._active then return false end
   for _, control in pairs(self._controls) do
-    control.value = 0
+    for i = 1, #control.values do
+      control.values[i] = 0
+    end
     local sources = control.sources[self._active]
     for i = 1, #sources do
       local source = sources[i]
-      control.value = control.value + source(self)
+      self:addSourceToValues(control.values, source)
+    end
+  end
```

Comment:
Alright, so you'll see that now _updateControls does three sequential loops:

1. Loop through keyboard / joystick controls and record raw input values.
2. Loop through aggregate controls (just axispairs right now) and update those values. This needs to happen separately after the keyboard / joystick controls, since it relies on those input values.
3. Loop through the final values for each control and clamp the value between 1 and -1, and record whether the input was "down".

Does this all make sense to you?